pkg/cluster/clustermsi.go

```diff
@@ -0,0 +1,145 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/msi-dataplane/pkg/dataplane"
+	"github.com/Azure/msi-dataplane/pkg/store"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/armmsi"
+)
+
+const (
+	mockMsiCertValidityDays = 90
+)
+
+// ensureClusterMsiCertificate leverages the MSI dataplane module to fetch the MSI's
+// backing certificate (if needed) and store the certificate in the cluster MSI key
+// vault. It does not concern itself with whether an existing certificate is valid
+// or not; that can be left to the certificate refresher component.
+func (m *manager) ensureClusterMsiCertificate(ctx context.Context) error {
+	secretName, err := m.clusterMsiSecretName()
+	if err != nil {
+		return err
+	}
+
+	_, err = m.clusterMsiKeyVaultStore.GetCredentialsObject(ctx, secretName)
+	if err == nil {
```

Comment:
I assume since we're not storing a var that the purpose of line 36 is to just check that the credentials object exists. If that's the case, do we need line 37 and 38 or should we just proceed to line 39/40?

Reply:
No, because if the credentials object already exists, we want to exit early instead of creating a new one and trying to replace the existing one.